src/lib/operators/window_function_evaluator/window_function_evaluator_traits.hpp

```diff
@@ -0,0 +1,319 @@
+#pragma once
+
+#include <limits>
+#include <type_traits>
+
+#include "expression/window_function_expression.hpp"
+#include "operators/aggregate/window_function_traits.hpp"
+#include "util.hpp"
+
+namespace hyrise::window_function_evaluator {
+
+constexpr static auto rank_like_window_functions =
+    std::array{WindowFunction::Rank, WindowFunction::DenseRank, WindowFunction::RowNumber, WindowFunction::PercentRank};
+
+constexpr bool is_rank_like(WindowFunction window_function) {
+  return std::ranges::find(rank_like_window_functions, window_function) != rank_like_window_functions.end();
+}
+
+namespace impls {
+
+template <typename Impl, typename OutputType>
+concept IsOnePassImpl =
+    requires {
+      typename Impl::State;
+      { Impl::initial_state(std::declval<const RelevantRowInformation&>()) } -> std::same_as<typename Impl::State>;
+      // NOLINTNEXTLINE(readability/braces)
```

Comment:
Add TODOs so that when one day the linter knows about C++20, the `NOLINT` s can be removed.